### PR TITLE
dbp-608-Fixed commit SHA for external Jobs

### PR DESCRIPTION
**Pull request by sahassou**
For external jobs, it is recommended to use fixed commit SHA references.

TODO:

Identify all external jobs currently referenced in our CI/CD workflows.
Update the workflow configurations to reference the external jobs using their respective commit SHAs instead of semantic versions.
AC:

All references to external jobs in the CI/CD workflows must use a fixed commit SHA.

# Description

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: .github/workflows/chart-release.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
           git config user.email "$GITHUB_ACTOR@users.noreply.github.com" 
 
       - name: Install Helm
-        uses: azure/setup-helm@5119fcb9089d432beecbf79bb2c7915207344b78 #v3.5
+        uses: azure/setup-kubectl@3e0aec4d80787158d308d7b364cb1b702e7feb7f #v4.0.0
```
Comment (aimee-889) [inline .github/workflows/chart-release.yaml:63]: has this willingly changed from helm to kubectl?

File: .github/workflows/chart-release.yaml [hunk 2]
```diff
@@ -60,7 +60,7 @@ jobs:
...
         with:
           version: 3.5.0
```
Comment (aimee-889) [inline .github/workflows/chart-release.yaml:65]: could you also increase the version to 3.12.* and test if it is still functional? We use that version in the devcontainer

File: .github/workflows/chart-release.yaml (hunk 1 above)
Comment (sahassou) [reply .github/workflows/chart-release.yaml]: I copied it to the wrong place, but I have corrected it

File: .github/workflows/chart-release.yaml (hunk 2 above)
Comment (sahassou) [reply .github/workflows/chart-release.yaml]: I have set up the correct azure/setup-helm; therefore, the version must be 3.5.0

Comment (aimee-889) [reply .github/workflows/chart-release.yaml]: i did not mean the version of the GHA, but the Helmversion

Comment (sahassou) [reply .github/workflows/chart-release.yaml]: wilcard can not be set as version, i get this erreur:Downloading v3.12.*
  Error: Error: Failed to download Helm from location https://get.helm.sh/helm-v3.12.*-linux-amd[6](https://github.com/dBildungsplattform/schulportal-client/actions/runs/7894948487/job/21546529374#step:4:7)4.zip

Therefore, I tested Helm version 3.12.3, and the pipeline successfully ran
https://github.com/dBildungsplattform/schulportal-client/actions/runs/7895165196

File: .github/workflows/check-trivy.yaml [hunk 3]
```diff
@@ -65,6 +65,6 @@ jobs:
 
       - name: Upload Trivy scan results to GitHub Security tab
         if: ${{ always() }}
-        uses: github/codeql-action/upload-sarif@0116bc2df50751f9724a2e35ef1f24d22f90e4e1 #v2.22.3
+        uses: github/codeql-action/upload-sarif@65c74964a9ed8c44ed9f19d4bbc5757a6a8e9ab9 #v2.16.1
```
Comment (aimee-889) [inline .github/workflows/check-trivy.yaml:68]: could you please recheck this? The latest release maybe called CodeQL Bundle, but under tags the other can be seen as well. In the repo they also say that version 2 is deprecated and will not get supported starting December 2024. Maybe it is good if we already use v3? See https://github.com/github/codeql-action?tab=readme-ov-file#supported-versions-of-the-codeql-action

Comment (sahassou) [reply .github/workflows/check-trivy.yaml]: codeql-action = v3.24.2
https://github.com/github/codeql-action/releases/tag/v3.24.2

**Review by aimee-889 — APPROVED** [review]

**Pull request merged**